packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -501,9 +501,26 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           // check @vite-ignore which suppresses dynamic import warning
           const hasViteIgnore = /\/\*\s*@vite-ignore\s*\*\//.test(rawUrl)
 
-          const url = rawUrl
+          let url = rawUrl
             .replace(/\/\*[\s\S]*?\*\/|([^\\:]|^)\/\/.*$/gm, '')
             .trim()
+
+          const [,
+            startQuotation,
+            importee,
+            endQuotation
+          ] = url.match(/^(.{1})(.*)(.{1})$/)!
+          const matchedAlias = config.resolve.alias.find(
+            (alias) => alias.find instanceof RegExp
+              ? alias.find.test(importee)
+              : importee.startsWith(alias.find)
+            )
+          if (matchedAlias) {
+            url = startQuotation +
+              importee.replace(matchedAlias.find, matchedAlias.replacement) +
+              endQuotation
+          }
```

Comment:
I don't think this is the right way to handle aliases. You can reference https://github.com/vitejs/vite/pull/6526 of how it implemented aliases in `import.meta.glob`, but the gist is to use Vite's resolver to resolve the path.

Some tests for this feature would be great too.